csrc/scheduler/matmul.cpp

```diff
@@ -354,46 +370,56 @@ void prologSwizzle(TensorView* shared_mem_tv, const MatmulParams& params) {
     //   -2   -1
     // [row, col]
     TORCH_INTERNAL_ASSERT(
-        tile_size_x % ldmatrix_rows == 0, "Partial matrices not supported");
-    shared_mem_tv->split(-2, ldmatrix_rows);
+        tile_size_x % n_rows == 0, "Partial matrices not supported");
+    shared_mem_tv->split(-2 - shift, n_rows);
     TORCH_INTERNAL_ASSERT(
-        tile_size_y % ldmatrix_cols == 0, "Partial matrices not supported");
-    shared_mem_tv->split(-1, ldmatrix_cols);
+        tile_size_y % n_cols == 0, "Partial matrices not supported");
+    shared_mem_tv->split(-1 - shift, n_cols);
     //     -4        -3      -2         -1
     // [matrix id, matrix, matrix id, matrix]
     TORCH_INTERNAL_ASSERT(
-        ldmatrix_rows % repeated_pattern_size == 0,
-        "ldmatrix_rows is assumed to be a multiple of repeated_pattern_size");
-    shared_mem_tv->split(-3, repeated_pattern_size);
+        n_rows % repeated_pattern_size == 0,
+        "n_rows is assumed to be a multiple of repeated_pattern_size");
+    if (repeated_pattern_size > 1) {
+      shared_mem_tv->split(-3 - shift, repeated_pattern_size);
+    }
     //     -5        -4      -3       -2         -1
     // [matrix id, repeat, pattern, matrix id, matrix]
-    int64_t swizzle_period = ldmatrix_rows / repeated_pattern_size;
-    TORCH_INTERNAL_ASSERT(
-        tile_size_y % (swizzle_period * ldmatrix_cols) == 0,
-        "need aperiodic swizzle config for tile size ",
-        tile_size_x,
-        "x",
-        tile_size_y,
-        "with units ",
-        ldmatrix_rows,
-        "x",
-        ldmatrix_cols);
-    shared_mem_tv->split(-2, swizzle_period);
+    int64_t swizzle_period = n_rows / repeated_pattern_size;
+    if (!shift) {
+      TORCH_INTERNAL_ASSERT(
+          tile_size_y % (swizzle_period * n_cols) == 0,
+          "need aperiodic swizzle config for tile size ",
+          tile_size_x,
+          "x",
+          tile_size_y,
+          "with units ",
+          n_rows,
+          "x",
+          n_cols);
+    }
+
+    shared_mem_tv->split(-2 - shift, swizzle_period);
     //     -6        -5      -4            -3           -2         -1
     // [matrix id, repeat, pattern, matrix id outer, pattern id, matrix]
     // swizzle repeat with pattern id to make repeat no longer repeat
-    if (isPowOf2(swizzle_period)) {
-      shared_mem_tv->swizzle(Swizzle2DType::XOR, -5, -2);
-    } else {
-      shared_mem_tv->swizzle(Swizzle2DType::CyclicShift, -5, -2);
+    if (!shift) {
```

Comment:
Why is this guarded with `if (!shift)`?

Reply:
I see why.

Should we do the following?
```C++
// TODO: This is a temporary workaround for the following issue:
// For the mma output, we have the following schedule:
// rFactor: [...., X, Y] -> mma-swizzle transformations -> leaf
// For epilogue smem tensor, the schedule is
// rFactor: [...., X, Y] -> split -> [...., X1, X2, X3, Y1, Y2, Y3] 
//   -> swizzle X2, Y2 -> [...., X1, X2', X3, Y1, Y2', Y3]
//   -> merge back -> [...., X', Y']
//   -> mma-swizzle transformations -> leaf
// The mma-swizzle transformations for the mma output and epilogue smem
// tensor are the same. In indexing, we do require {X, X'} and {Y, Y'} to be
// mapped in CA map, however, we currently can not handle that. So we have to
// do the same split and merge to the mma output without actually applying the
// swizzle, and this check is to detect and handle this specific case. We should
// remove this special handling when we fix our CA mapping.
if (shared_mem_tv->getMemoryType() != MemoryType::Shared) {
```
I feel that this is easier to read.